src/utils/simulateTransactions.ts

```diff
@@ -2,18 +2,21 @@ import { CreditPosition, OPENED_STATUS } from '@src/core/types';
 
 import { toWei } from './format';
 
+// TODO: replace all instances of Number with BigNumber
 export const borrowUpdate = (position: CreditPosition, amount: string) => {
   const borrowedAmount = toWei(amount, Number(+position['token'].decimals));
   const updatedPrincipal = Number(position['principal']) + Number(borrowedAmount);
   return { ...position, principal: `${updatedPrincipal}` };
 };
 
+// TODO: replace all instances of Number with BigNumber
 export const withdrawUpdate = (position: CreditPosition, amount: string) => {
```

Comment:
pro move would be to create a higher order function to abstract away a lot of this boilerplate e.g.
```
const updatePosition(updateFunc) => (position) => ({ ...position, ...updateFunc(position)})`
const depositAndRepayUpdate = updatePosition(() => { /* your update logic here */} )
```

Just need to return a valid Position from your update func

Reply:
This is legacy code not added to PR. 

This is a TODO refactoring task. Added to [Notion task](https://www.notion.so/debtdao/Abstract-Boilerplate-Frontend-Code-for-Simulating-Transactions-21de1ad57f9d460582ebb95960155b5f)